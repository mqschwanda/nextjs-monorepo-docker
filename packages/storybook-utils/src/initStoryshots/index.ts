import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
