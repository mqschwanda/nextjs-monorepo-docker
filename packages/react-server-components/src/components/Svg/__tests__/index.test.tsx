import { render } from '@testing-library/react';
import path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { Svg } from '@';

initStoryshots({
  configPath: path.resolve(__dirname, '../../../../../storybook/.storybook'),
  storyKindRegex: /^@mqs\/react-server-components\/components\/Svg$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<Svg />', () => {
      it('renders', () => {
        const testId = Svg.name;

        const { getByTestId, asFragment } = render((
          <Svg
            className='test'
            testId={testId}
          >
            <path
              d='M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z'
              strokeLinecap='round'
              strokeLinejoin='round'
              strokeWidth='2'
            />
          </Svg>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
