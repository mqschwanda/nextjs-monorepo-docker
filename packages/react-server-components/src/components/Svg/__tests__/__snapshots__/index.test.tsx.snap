// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <Svg /> renders 1`] = `
<DocumentFragment>
  <svg
    class="fill-inherit stroke-inherit stroke-1 test"
    data-testid="Svg"
    fill="currentColor"
    stroke="currentColor"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/Svg Svg Example 1`] = `
<svg
  className="fill-none stroke-current stroke-1 h-6 w-6"
  fill="currentColor"
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
  />
</svg>
`;
